### PR TITLE
feat(RegularGrid3D): add support for .vo

### PR DESCRIPTION
**Pull request by BotellaA**

### CODE REVIEW

File: src/geode/io/geosciences/vo_input.cpp [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+        void initialize_grid()
+        {
+            auto line = geode::detail::goto_keyword( file_, "AXIS_O" );
+            auto origin = read_coord( line, 1 );
+            const auto grid_size = read_grid_size( origin );
+            auto cells_number = read_cells_number();
+            auto cells_length = compute_cells_length( grid_size, cells_number );
+
+            builder_->initialize_grid(
+                origin, std::move( cells_number ), std::move( cells_length ) );
```
Comment (BotellaA) [inline src/geode/io/geosciences/vo_input.cpp:75]:
```suggestion
                std::move( origin ), std::move( cells_number ), std::move( cells_length ) );
```

File: src/geode/io/geosciences/vo_input.cpp [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+            std::array< double, 3 > result;
+            for( const auto axis_id : geode::LRange{ 3 } )
+            {
+                result[axis_id] = grid_size[axis_id] / nb_cells[axis_id];
+            }
+            return result;
+        }
+
+        geode::Point3D read_coord(
+            absl::string_view line, geode::index_t offset ) const
```
Comment (BotellaA) [inline src/geode/io/geosciences/vo_input.cpp:115]: This is not already done in this project?

File: src/geode/io/geosciences/vo_input.cpp [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+            return { { geode::string_to_double( tokens[offset] ),
+                geode::string_to_double( tokens[1 + offset] ),
+                geode::string_to_double( tokens[2 + offset] ) } };
+        }
+
+        void read_data_file()
+        {
+            auto line = geode::detail::goto_keyword( file_, "ASCII_DATA_FILE" );
+            auto tokens = geode::string_split( line );
+            DEBUG( tokens[1] );
```
Comment (BotellaA) [inline src/geode/io/geosciences/vo_input.cpp:129]:
```suggestion
```

File: src/geode/io/geosciences/vo_input.cpp [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+            std::ifstream data_file{ absl::StrCat( file_folder_, tokens[1] ) };
+
+            std::getline( data_file, line );
+            std::getline( data_file, line );
+            DEBUG( line );
```
Comment (BotellaA) [inline src/geode/io/geosciences/vo_input.cpp:134]:
```suggestion
```

File: src/geode/io/geosciences/vo_input.cpp [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+            tokens = geode::string_split( line );
+            std::vector< std::shared_ptr< geode::VariableAttribute< double > > >
+                data_attributes;
+            geode::index_t name_id{ 4 };
+            while( name_id < tokens.size() )
+            {
+                DEBUG( tokens[name_id] );
```
Comment (BotellaA) [inline src/geode/io/geosciences/vo_input.cpp:141]:
```suggestion
```

File: src/geode/io/geosciences/vo_input.cpp [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+            auto line = geode::detail::goto_keyword( file_, "ASCII_DATA_FILE" );
+            auto tokens = geode::string_split( line );
+            DEBUG( tokens[1] );
+            std::ifstream data_file{ absl::StrCat( file_folder_, tokens[1] ) };
+
+            std::getline( data_file, line );
+            std::getline( data_file, line );
+            DEBUG( line );
+            tokens = geode::string_split( line );
+            std::vector< std::shared_ptr< geode::VariableAttribute< double > > >
```
Comment (BotellaA) [inline src/geode/io/geosciences/vo_input.cpp:136]: Could it be a FixedArray of size ̀tokens.size() - 4`?

File: src/geode/io/geosciences/vo_input.cpp (hunk 2 above)
Comment (MelchiorSchuh) [reply src/geode/io/geosciences/vo_input.cpp:115]: It is, I can homogenize it by adding ti to the utils.cpp

**Review by MelchiorSchuh — APPROVED** [review]

**Pull request merged**